src/plugins/world_control/WorldControl.cc

```diff
@@ -238,33 +256,21 @@ void WorldControl::OnWorldStatsMsg(const ignition::msgs::WorldStatistics &_msg)
 /////////////////////////////////////////////////
 void WorldControl::OnPlay()
 {
-  std::function<void(const ignition::msgs::Boolean &, const bool)> cb =
-      [this](const ignition::msgs::Boolean &/*_rep*/, const bool _result)
-  {
-    if (_result)
-      QMetaObject::invokeMethod(this, "playing");
-  };
-
-  ignition::msgs::WorldControl req;
-  req.set_pause(false);
+  ignition::msgs::WorldControl msg;
+  msg.set_pause(false);
   this->dataPtr->pause = false;
-  this->dataPtr->node.Request(this->dataPtr->controlService, req, cb);
```

Comment:
`ign-gazebo` is not the only application using these plugins, so we should be mindful of other downstream users when changing behaviours. Can we do both (service + event), and offer the user an option to disable one or the other via XML?

Reply:
Yeah, thanks for the reminder - I have added support for event or service in https://github.com/ignitionrobotics/ign-gui/pull/306/commits/7ff16ce47ffbb1e0dc37d005333a4111a13e505d. The following can be added to `ign-gazebo`'s `gui.config` if you want to use a service (event is now the default):

```diff
diff --git a/src/gui/gui.config b/src/gui/gui.config
index 36166061..28bfedf9 100644
--- a/src/gui/gui.config
+++ b/src/gui/gui.config
@@ -137,6 +137,7 @@
   <step>true</step>
   <start_paused>true</start_paused>
 
+  <use_event>false</use_event>
 </plugin>
 
 <!-- World statistics -->
```

Does the migration guide need to be updated? If so, is that an update that should take place here, or in `ign-gazebo`?

Reply:
I changed the default to use the service so that we don't break downstream users with a behavior change. On the Gazebo side, I set the default gui configs to set `<use_event>true</use_event>` in https://github.com/ignitionrobotics/ign-gazebo/pull/1109/commits/ae91a9dab88d717124de89a1fd6d1ee5a2045e3b

Reply:
> I changed the default to use the service so that we don't break downstream users with a behavior change. On the Gazebo side, I set the default gui configs to set `<use_event>true</use_event>`

That sounds good, thanks for making the change. Since behavior is still preserved for downstream users, I think we can mark this as resolved.